apps/OpenSpace/ext/launcher/src/sgctedit/filesupport.cpp

```diff
@@ -0,0 +1,278 @@
+
+FileSupport::FileSupport(QVBoxLayout* parentLayout, std::vector<QRect>& monitorList,
+                         Display* display, Orientation* orientation,
+                         std::vector<sgct::config::Window>& windowList,
+                         sgct::config::Cluster& cluster, std::function<void(bool)> cb)
+    : _displayWidget(display)
+    , _orientationWidget(orientation)
+    , _monitors(monitorList)
+    , _cluster(cluster)
+    , _windowList(windowList)
+    , _finishedCallback(cb)
+{
+    QVBoxLayout* layoutFullVertical = new QVBoxLayout();
+    _lineFilename = new QLineEdit();
+    _lineFilename->setFixedWidth(190);
+    {
+        QHBoxLayout* layoutFilename = new QHBoxLayout();
+        QLabel* labelFilename = new QLabel();
+        labelFilename->setText("Filename: ");
+        layoutFilename->addStretch(1);
+        layoutFilename->addWidget(labelFilename);
+        layoutFilename->addWidget(_lineFilename);
+        layoutFilename->addStretch(1);
+        layoutFullVertical->addLayout(layoutFilename);
+    }
+    _saveButton = new QPushButton("Save");
+    _saveButton->setToolTip("Save global orientation changes");
+    connect(_saveButton, SIGNAL(released()), this, SLOT(save()));
+    _saveButton->setEnabled(false);
+    _cancelButton = new QPushButton("Cancel");
+    _cancelButton->setToolTip("Cancel global orientation changes");
+    connect(_cancelButton, SIGNAL(released()), this, SLOT(cancel()));
+    {
+        QHBoxLayout* layoutButtonBox = new QHBoxLayout;
+        layoutButtonBox->addStretch(1);
+        layoutButtonBox->addWidget(_saveButton);
+        layoutButtonBox->addWidget(_cancelButton);
+        layoutFullVertical->addLayout(layoutButtonBox);
+    }
+    parentLayout->addLayout(layoutFullVertical);
+    connect(_lineFilename, SIGNAL(textEdited(const QString&)), this,
+        SLOT(filenameEdited(const QString&)));
+}
+
+void FileSupport::saveCluster() {
+    if (_orientationWidget) {
+        sgct::config::Scene initScene;
+        initScene.orientation = _orientationWidget->orientationValue();
+        if (_cluster.nodes.size() == 0) {
+            _cluster.nodes.push_back(sgct::config::Node());
+        }
+        _cluster.masterAddress = "localhost";
+        _cluster.nodes.back().address = "localhost";
+        _cluster.nodes.back().port = 20401;
+        _cluster.scene = std::move(initScene);
+        _cluster.firmSync = _orientationWidget->vsyncValue();
+    }
+}
+
+void FileSupport::saveUser() {
+    if (_orientationWidget) {
+        sgct::config::User user;
+        user.eyeSeparation = 0.065;
+        user.position = {0.0, 0.0, 4.0};
+        _cluster.users.push_back(user);
+    }
+}
+
+bool FileSupport::isWindowFullscreen(unsigned int monitorIdx, sgct::ivec2 wDims) {
+    sgct::ivec2 mDims = {_monitors[monitorIdx].width(), _monitors[monitorIdx].height()};
+    return ((mDims.x == wDims.x) && (mDims.y == wDims.y));
+}
+
+bool FileSupport::findGuiWindow(unsigned int& foundWindowIndex) {
```

Comment:
This would be a nice way to do it, but not easier. A bool is easy and the caller is a bool with a name that explains what the bool condition means.

Reply:
Just a general dislike of `out` parameters in functions, since they don't really serve any purpose anymore.   The std-library way of having it would be returning a `std::pair<unsigned int, bool>` like (std::map::emplace) but I think std::optional<unsigned int> would signal the intent of the function that it is possible to fail much clearer than either that or using a sentiel value